modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -1506,6 +1518,18 @@ Optional<Long> highestRevision(boolean addition) {
                     .map(Map.Entry::getKey);
         }
 
+        /**
+         * Returns the highest revision which is presented in the {@link ZoneState#topologyAugmentationMap()}.
+         *
+         * @return The highest revision which is presented in the {@link ZoneState#topologyAugmentationMap()}.
+         */
+        Optional<Long> highestRevision() {
```

Comment:
I know a guy who would say that we should use OptionalLong here))

Reply:
It needs to convert stream here: from Stream<Long> to LongStream or something like that. Looks like solution is worse than the "trouble" (

Reply:
I'm not sure whether mapToLong() is worth than Comparator.naturalOrder(). Up to you.